### PR TITLE
Add feature gate for policy preemption

### PR DESCRIPTION
**Pull request by Poor12**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add feature gate for policy preemption.

**Which issue(s) this PR fixes**:
Part of #3787 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
None

### CODE REVIEW

**Review by whitewindmills — COMMENTED** [review]
/lgtm

File: pkg/features/features.go [hunk 1]
```diff
@@ -18,6 +18,9 @@ const (
 
 	// CustomizedClusterResourceModeling indicates if enable cluster resource custom modeling.
 	CustomizedClusterResourceModeling featuregate.Feature = "CustomizedClusterResourceModeling"
+
+	// PolicyPreemption indicates if high-priority policies could preempt resource templates which are matched by low-priority policies.
+	PolicyPreemption featuregate.Feature = "PolicyPreemption"
```
Comment (RainbowMango) [inline pkg/features/features.go:23]:
```suggestion
	PolicyPreemption featuregate.Feature = "PropagationPolicyPreemption"
```
As I commented at https://github.com/karmada-io/karmada/pull/3684#discussion_r1259416715, we'd better note the feature gate more clearly.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**